tests/test_edgeql_ddl.py

```diff
@@ -2642,6 +2642,141 @@ async def test_edgeql_ddl_function_29(self):
                 CREATE TYPE test::foo;
             ''')
 
+    async def test_edgeql_ddl_function_rename_01(self):
+        await self.con.execute("""
+            CREATE FUNCTION test::foo(s: str) -> str {
+                USING (SELECT s)
+            }
+        """)
+
+        await self.assert_query_result(
+            """SELECT test::foo("a")""",
+            ["a"],
+        )
+
+        await self.con.execute("""
+            ALTER FUNCTION test::foo(s: str)
+            RENAME TO test::bar;
+        """)
+
+        await self.assert_query_result(
+            """SELECT test::bar("a")""",
+            ["a"],
+        )
+
+        await self.con.execute("""
+            DROP FUNCTION test::bar(s: str)
+        """)
+
+    async def test_edgeql_ddl_function_rename_02(self):
+        await self.con.execute("""
+            CREATE FUNCTION test::foo(s: str) -> str {
+                USING (SELECT s)
+            };
+
+            CREATE FUNCTION test::bar(s: int64) -> str {
+                USING (SELECT <str>s)
+            };
+        """)
+
+        with self.assertRaisesRegex(
+                edgedb.SchemaError,
+                r"can not rename function to 'test::foo' because "
+                r"it already exists"):
+            await self.con.execute("""
+                ALTER FUNCTION test::bar(s: int64)
+                RENAME TO test::foo;
+            """)
+
+    async def test_edgeql_ddl_function_rename_03(self):
+        await self.con.execute("""
+            CREATE FUNCTION test::foo(s: str) -> str {
+                USING (SELECT s)
+            };
+
+            CREATE FUNCTION test::foo(s: int64) -> str {
+                USING (SELECT <str>s)
+            };
+        """)
+
+        with self.assertRaisesRegex(
+                edgedb.SchemaError,
+                r"renaming an overloaded function is not allowed"):
+            await self.con.execute("""
```

Comment:
So how would I do rename 2 or more functions sharing the same name? Recreate all functions from scratch?

Reply:
Yeah, unfortunately.

Reply:
Can we allow a special form of syntax like `ALTER FUNCTION foo::bar RENAME TO foo::baz` -- note that there are no parenthesis there. @elprans

Reply:
Actually, I think we should allow parentheses-free form to refer to a _single_, non-overloaded function, like Postgres does.  The vast majority of user-defined functions are not overloaded, and having to retype full parameter definition is an unnecessary hassle (which is why Postgres implemented this).

Reply:
I'm not against that. But if we have more than one function defined for the name, would it make sense to allow renaming all of them with this syntax (I think it would)?

Reply:
Would `DROP FUNCTION foo::bar` drop all overloads too?

Reply:
That's a good question.  Probably that would be too dangerous. Maybe we should have `ALTER ALL FUNCTION foo::bar RENAME TO` and `DROP ALL FUNCTION foo::bar`